packages/dos-donts-block/src/DoDontItem.tsx

```diff
@@ -1,89 +1,309 @@
-                    <div
-                        data-test-id="dos-donts-icon"
-                        className={joinClassNames(['tw-mr-2 tw-w-auto', shouldBlurIcon ? 'tw-opacity-30' : ''])}
-                    >
-                        {type === DoDontType.Do && <IconCheckMarkCircle size={IconSize.Size24} />}
-                        {type === DoDontType.Dont && <IconCrossCircle size={IconSize.Size24} />}
-                    </div>
-                )}
-                <div className="tw-w-full">
-                    <RichTextEditor
-                        designTokens={designTokens ?? undefined}
-                        value={title}
-                        onTextChange={(value) => saveItem(id, value, 'title')}
-                        placeholder={editing ? 'Add a title' : ''}
-                        readonly={!editing}
-                    />
-                </div>
-            </div>
-            {style === DoDontStyle.Underline && (
-                <hr
-                    style={dividerStyles[type]}
-                    className="tw-w-full tw-mt-4 tw-mb-5 tw-h-1 tw-border-none tw-rounded tw-bg-black-40"
+        const onUploadClick = () => {
+            openFileDialog();
+        };
+
+        useEffect(() => {
+            if (selectedFiles) {
+                setIsUploadLoading(true);
+                uploadFile(selectedFiles);
+            }
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+        }, [selectedFiles]);
+
+        useEffect(() => {
+            if (doneAll) {
+                (async (uploadResults) => {
+                    const imageId = uploadResults?.[0]?.id;
+                    const existingIds = itemImages?.map((x) => x.id) || [];
+                    const newIds = [...new Set([...existingIds, imageId])];
+                    updateAssetIdsFromKey('itemImages', newIds).then(() => {
+                        setIsUploadLoading(false);
+                        onChangeItem(id, imageId, 'imageId');
+                    });
+                })(uploadResults);
+            }
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+        }, [doneAll, uploadResults]);
+
+        const isEmpty = (text: string): boolean => {
+            if (text && text !== '') {
+                return JSON.parse(text).every((element: EditorElement) => {
+                    const elementsWithText = element.children.filter((child: EditorChild) => child.text !== '');
+                    return elementsWithText.length === 0;
+                });
+            }
+            return true;
+        };
+
+        const shouldBlurIcon = useMemo(() => isEmpty(title), [title]);
+
+        const titleRichTextEditor = useMemo(() => {
+            return (
+                <RichTextEditor
+                    designTokens={designTokens ?? undefined}
+                    value={title}
+                    onTextChange={(value) => debounce(() => onChangeItem(id, value, 'title'), 500)()}
```

Comment:
`onTextChange` from the RichTextEditor is already debounce, don't think it's necessary to do it here again. Also maybe it would make sense to save the item also `onBlur`, @mike-schmid just did that in the other blocks in this PR: https://github.com/Frontify/guideline-blocks/pull/446